### PR TITLE
Coco JR marketing page

### PR DESCRIPTION
**Pull request by adamkecskes**
Live on [next](https://next.codecombat.com/junior)
| desktop | mobile | 
| --- | --- | 
| localhost_3000_junior | localhost_3000_junior (1) |

## Release Notes

- **New Features**
  - Introduced a new `aspectRatio` property for video components, enhancing flexibility in video display.
  - Added a new route and view for the "junior" coding curriculum, including a dedicated landing page with engaging content.
  - Implemented a testimonials carousel to showcase user feedback on the junior curriculum.
  - Launched a `TrendsAndInsights` component tailored for a junior audience, providing relevant educational trends.

- **Content Enhancements**
  - Expanded localization support with new strings related to the junior curriculum, improving informational depth and marketing appeal.

- **Style Improvements**
  - Increased dimensions and border-radius of avatar elements for improved visual appeal.
  - Added underlining for links in the MixedColorLabel component to enhance visibility.
  - Enhanced header responsiveness for better visual presentation on smaller devices.
  - Standardized formatting by adding trailing commas in various data structures for improved code consistency.

- **Bug Fixes**
  - Adjusted modal visibility initialization to ensure proper display upon component mounting.
  - Modified carousel navigation to prevent display of dots when only one item is present.

### CODE REVIEW

File: app/locale/en.js [hunk 1]
```diff
@@ -6233,6 +6233,43 @@ module.exports = {
       get_full_course: 'Get Full Course',
       try_sample_lesson: 'Try Sample Lesson',
       trophy_header: 'Level up as a Computer Science Teacher with our Professional Development offerings!'
+    },
+    junior_page: {
+      testimonials_1_quote: 'The future is programmable and you will need to be able to code to participate…I think CoCo [CodeCombat] has the best CS curriculum on the market… the teacher tool kit is what sets it apart.',
```
Comment (nwinter) [inline app/locale/en.js:6238]: We can fix up the formatting a bit here.
```suggestion
      testimonials_1_quote: 'The future is programmable and you will need to be able to code to participate… I think CodeCombat has the best CS curriculum on the market… the teacher toolkit is what sets it apart.',
```

File: app/locale/en.js [hunk 2]
```diff
@@ -6233,6 +6233,43 @@ module.exports = {
...
+      testimonials_1_name: 'Jacob Kaminski',
+      testimonials_1_title: 'STEM Program Director, American STEM Prep',
+      start_their_coding_journey_today: 'Start their **coding journey** today!',
+      our_flagship_k_5_curriculum: 'Our flagship K-5 curriculum features a scaffolded progression of learning levels that teach **coding concepts** at a slower pace, modeled specifically for elementary students.',
+      sign_up_to_start_today: 'Sign up to start today',
+      testimonials_1_full_review_text: 'Full review text',
+      scaffolded_approach: 'Scaffolded Approach',
+      no_matter_the_coding_experience: 'No matter the coding experience, **CodeCombat Junior** meets students where they’re at. They can code with icon blocks, word blocks, blocks and text code in parallel, or text-based coding with industry standard languages.',
```
Comment (nwinter) [inline app/locale/en.js:6246]:
```suggestion
      no_matter_the_coding_experience: 'No matter the coding experience, **CodeCombat Junior** meets students where they’re at. They can program with icon blocks, word blocks, blocks and text code in parallel, or text-based code with industry-standard languages like Python and JavaScript.',
```

File: app/locale/en.js [hunk 3]
```diff
@@ -6233,6 +6233,43 @@ module.exports = {
...
+      check_out_this_resource: 'Check out **[https://docs.google.com/spreadsheets/d/1_2zr55qsEcbHDyDNoelYamN9ov0G35ju7s-ufttSzVQ]this resource** as a starting point for your personalized approach.',
+      connect_with_our_team: 'Connect with our team to access all the teacher resources, a personalized learning pathway, generative practice levels, and more. We’ll help you integrate this comprehensive K-5 coding curriculum to ensure your students thrive at every stage of their coding journey.',
+      faq_1_question: 'FAQ 1 Question',
+      faq_1_answer: 'FAQ 1 Answer',
+      faq_2_question: 'FAQ 2 Question',
+      faq_2_answer: 'FAQ 2 Answer',
+      faq_3_question: 'FAQ 3 Question',
+      faq_3_answer: 'FAQ 3 Answer',
+      faq_4_question: 'FAQ 4 Question',
+      faq_4_answer: 'FAQ 4 Answer',
```
Comment (nwinter) [inline app/locale/en.js:6264]: This content is coming, right?

File: app/locale/en.js [hunk 4]
```diff
@@ -6233,6 +6233,43 @@ module.exports = {
...
+      meta_title: 'Meta Title',
```
Comment (nwinter) [inline app/locale/en.js:6265]:
```suggestion
      meta_title: 'CodeCombat Junior: K-5 Computer Science for Schools',
```

**Review by nwinter — CHANGES_REQUESTED** [review]
Visually, the page looks great! Found a bunch of minor stuff.

The blue bolded "CodeCombat Junior" elements throughout the page look like links but aren't links. I guess this is the same issue as in these elements on the /schools page, but those aren't as prominent (they don't start until the AI HackStack section). Maybe here, and there, we can adjust the design slightly to avoid collision with links?

_Can you tell which of these are links?_
Screenshot 2024-09-19 at 08 43 29
Screenshot 2024-09-19 at 08 48 53
Screenshot 2024-09-19 at 08 49 02
Screenshot 2024-09-19 at 08 49 09
Screenshot 2024-09-19 at 08 49 17
Screenshot 2024-09-19 at 08 49 26

Mobile version has a bunch of dead white space at the top of the page. (This is also true of other landing pages.) Is this intentional? I would think we would pull all lthat up, or at least put a cool logo there or something.
CodeCombat the most effective computer science solution for schools

Lighthouse on marketing pages is important; checked it for desktop real quick, looks ok, although there are a few things to fix. You can run it yourself to see their list. Doesn't need to be perfect, but check for low-hanging fruit.
Screenshot 2024-09-19 at 09 43 19
I think the biggest question from this is what to do about the Cloudflare video at the top, which loads a bunch of data. If it impacts Lighthouse scores, is there a good way to defer it such that it doesn't?

For all fixes, consider doing them on all the other pages built out of similar components or methods.

**Review by mrfinch — APPROVED** [review]

**Pull request merged**